meson.build

```diff
@@ -22,7 +22,16 @@ if build_machine.system() != 'windows'
     default_warning_flags += ['-Wno-empty-body']
   endif
   default_warning_flags += ['-Wno-unused-parameter',
-                            '-Wno-deprecated-declarations']
+                            '-Wno-deprecated-declarations',
+                            '-Wno-error=cast-function-type',
+                            '-Wno-cast-function-type',
+                            '-Wno-error=stringop-truncation',
+                            '-Wno-stringop-truncation',
+                            '-Wno-error=missing-field-initializers',
+                            '-Wno-missing-field-initializers',
+                            '-Wno-error=class-memaccess',
+                            '-Wno-class-memaccess'
```

Comment:
Some of those exclusions are for "external" code like openmpi but some of them are for internal/SU2 code and code be removed after fixing the issues.